app/observers/article_observer.rb

```diff
@@ -1,16 +1,19 @@
 class ArticleObserver < ApplicationObserver
-  def after_save(article)
+  def after_create(article)
```

Comment:
This would need to remain `after_save` because users often don't publish their articles upon first creating them. That's what the logic below was doing. I think this started creating too much noice when the `warned_user_ping` was introduced without this logic about whether it was just published.

FWIW the existing logic, while _okay_ wasn't great. Using the logic we use for notifications probably makes more sense:

`if @article.published && @article.saved_changes["published_at"]&.include?(nil)`

This checks if it was published and the `published_at` was also just changed. Though I'm not sure we have access to this in an observer context.

Reply:
Left a note below.